src/MessageTrait.php

```diff
@@ -105,35 +89,8 @@ public function withHeader($header, $value): self
 
     public function withAddedHeader($header, $value): self
     {
-        if (!is_array($value)) {
-            $value = [$value];
-        } elseif (!empty($value)) {
-            $value = array_values($value);
-        } else {
-            throw new \InvalidArgumentException('Header values must be strings');
-        }
-
-        if (!is_string($header) || empty($header)) {
```

Comment:
Cannot remove this. 
What if $header is an object? We cannot use objects as array keys.

Reply:
Yeah, I missed your comment and for some reason I didn’t realise that was what was happening. I’ll put it back!